docs/sql/data_types.txt

```diff
@@ -170,6 +170,74 @@ Create table example::
     CREATE OK (... sec)
 
 
+.. _geo_shape_data_type:
+
+geo_shape
+=========
+
+The ``geo_shape`` type is used to store geometric shapes as defined in
+`GeoJSON geometry objects`_.
+
+A geo_shape column can store different kinds of `GeoJSON geometry objects`_.
+Thus it is possible to store e.g. ``LineString`` and ``MultiPolygon`` shapes
+into the same column.
+
+Definition
+----------
+
+A geo_shape Column is defined using the following Syntax::
+
+    <columnName> GEO_SHAPE [ (<parameter>=<value> [, ...]) ]
+
+Where the following parameters are supported:
+
+
+:precision: (Default: ``50m``) Define the precision of the indexed shape used for querying.
+            Given as string containing a number and an optional distance unit (defaults to ``m``).
+
+            Supported units are ``inch`` (``in``), ``yard`` (``yd``), ``miles`` (``mi``),
+            ``kilometers`` (``km``), ``meters``(``m``), ``centimeters`` (``cm``), ``millimeters`` (``mm``).
+
+:tree: (Default: ``geohash``) Internal data structure used for indexing.
+       Available trees are ``geohash`` and ``quadtree``.
+       Both are prefix trees representing the earth as grid layers with increasing precision.
+       ``quadtree`` uses a `Quadtree`_ to represent grids and ``geohash`` uses `Geohash`_.
+       The main difference is that the ``quadtree`` supports higher precision
+       than the ``geohash`` tree.
```

Comment:
i think the tree should be explained more detailed. How does it affect index size? Advantages and disadvantages of quadtree / geohash